app/components/framework/FrameworkImageButton/index.tsx

```diff
@@ -44,15 +59,33 @@ function FrameworkImageButton(props: Props) {
         hideModal,
     ] = useModalState(false);
 
+    const analysisFrameworkVariables = useMemo(
+        (): AnalysisFrameworkQueryVariables | undefined => (
+            frameworkId ? { frameworkId } : undefined
+        ),
+        [frameworkId],
+    );
+
     const {
-        pending,
-        response: frameworkDetails,
-    } = useRequest<AnalyticalFramework>({
-        skip: !isModalVisible || !frameworkId,
-        url: `server://analysis-frameworks/${frameworkId}/`,
-        query,
-        method: 'GET',
-    });
+        data: analysisFrameworkData,
+        loading: analysisFrameworkLoading,
+    } = useQuery<AnalysisFrameworkQuery, AnalysisFrameworkQueryVariables>(
+        ANALYSIS_FRAMEWORK,
+        {
+            skip: isNotDefined(analysisFrameworkVariables),
+            variables: analysisFrameworkVariables,
+        },
+    );
+
+    const imageSource = useMemo(
+        () => {
+            if (isNotDefined(analysisFrameworkData?.analysisFramework?.previewImage?.url)) {
+                return image;
+            }
+            return analysisFrameworkData?.analysisFramework?.previewImage?.url;
```

Comment:
This logic doesn't look right. Can you please recheck?

Reply:
I think its good, if image url is not present we have to send images from props.